tests/conftest.py

```diff
@@ -9,3 +13,11 @@ def disable_contract_checking():
     python_ta.contracts.ENABLE_CONTRACT_CHECKING = False
     yield
     python_ta.contracts.ENABLE_CONTRACT_CHECKING = True
+
+
+@pytest.fixture()
+def prevent_webbrowser_and_httpserver():
```

Comment:
It's great that you've identified the relevant mocking code. Pulling it out into a fixture is a good idea, but this is revealing another problem with this code: it doesn't "reset" the functions to the original behaviour after the test case ends. This defeats the purpose of mocking, which should be tied to specific test cases without changing any global behaviour.

You can take this opportunity to add `pytest-mock` (https://github.com/pytest-dev/pytest-mock/) as a development dependency for the project, and then use it to set up this mocking. That library should correctly handle restoring the original behaviour after the tests end.